### PR TITLE
Fix api server discovery

### PR DESCRIPTION
**Pull request by QuentinBisson**
Towards https://github.com/giantswarm/giantswarm/issues/22500

Tested on ghost MC, waiting for 3.7.0 to be released before testing on a workload cluster.

## Checklist

I have:

- [x] Described why this change is being introduced
- [x] Separated out refactoring/reformatting in a dedicated PR
- [x] Updated changelog in `CHANGELOG.md`

### CODE REVIEW

File: files/templates/scrapeconfigs/additional-scrape-configs.template.yaml [hunk 1]
```diff
@@ -9,8 +9,6 @@
   - source_labels: [__meta_kubernetes_service_label_component]
     regex: apiserver
     action: keep
-  - target_label: __address__
```
Comment (QuentinBisson) [inline files/templates/scrapeconfigs/additional-scrape-configs.template.yaml:12]: This allows prometheus to scrape all 3 api servers (not limited to 1 master)

File: service/key/key.go [hunk 2]
```diff
@@ -201,7 +201,7 @@ func AlertManagerKey() string {
 func APIUrl(obj interface{}) string {
 	switch v := obj.(type) {
 	case *v1.Service:
-		return fmt.Sprintf("%s:443", v.Spec.ClusterIP)
+		return "kubernetes.default:443"
```
Comment (QuentinBisson) [inline service/key/key.go:204]: This makes us less reliant on the load balancer IP that can (albeit not often) change

**Comment by QuentinBisson** [comment]
Tested on Gauss for both WC and MCs

File: service/key/key.go (hunk 2 above)
Comment (whites11) [reply service/key/key.go:204]: i doubt it can ever change, but I still think it's worth the change

File: CHANGELOG.md [hunk 3]
```diff
@@ -12,10 +12,26 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
 
+### Fixed 
+
+- Fix API server discovery.
+
 ### Removed
 
 - Remove duplicate scrape config targets.
 
+### Fixed
```
Comment (whites11) [inline CHANGELOG.md:23]: How come we have this in the diff?

Comment (QuentinBisson) [reply CHANGELOG.md]: Fixed

**Review by whites11 — APPROVED** [review]

**Pull request merged**